pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -78,19 +86,43 @@ func (p *Plugin) GetResultType() string {
 	return p.Definition.ResultType
 }
 
+//FillTemplate populates the internal Job YAML template with the values for this particular job.
+func (p *Plugin) FillTemplate(hostname string) ([]byte, error) {
+	var b bytes.Buffer
+	// TODO (EKF): Should be YAML once we figure that out
+	container, err := json.Marshal(&p.Definition.Spec)
+	if err != nil {
```

Comment:
we should be consistent about `if err := fun(); err != nil` or `err := fun(); if err != nil` at least within the same function.

Reply:
you can only do the latter if error is the only return, because the variables are scoped to the if statement. Unless I'm missing something?

Reply:
nah u right